apps/src/JSInterpreter.js

```diff
@@ -324,6 +326,17 @@ JSInterpreter.prototype.removeStoppedAtBreakpointRowForScope = function (scope)
 };
 
 /**
+ * Determines if the program is done executing.
+ *
+ * @return {boolean} true if program is complete (or an error has occurred).
+ */
+JSInterpreter.prototype.isProgramDone = function () {
+  return this.executionError ||
+      !this.interpreter ||
+      !this.interpreter.stateStack.length;
+};
```

Comment:
`try` / `catch` is coming: https://github.com/code-dot-org/code-dot-org/pull/6927

We may be able to alter this a little after we merge that PR.